pkgs/development/tools/database/sqlcmd/default.nix

```
@@ -0,0 +1,55 @@
+{ buildGoModule
+, fetchFromGitHub
+, installShellFiles
+, lib
+, sqlcmd
+, testers
+}:
+
+buildGoModule rec {
+  pname = "sqlcmd";
+  version = "0.15.3";
+
+  src = fetchFromGitHub {
+    repo = "go-sqlcmd";
+    owner = "microsoft";
+    rev = "v${version}";
+    sha256 = "sha256-1CdZVh7pbupCNOE1ydgYEqX4rYo2LRddlNRNf0QjlN0=";
+  };
+
+  vendorSha256 = "sha256-6JfxKzYAjSQ9JFuFGDUZ0ALS1D7f2LK3bP0Fbl2ivo0=";
+  proxyVendor = true;
+
+  ldflags = [ "-s" "-w" "-X main.version=${version}" ];
+
+  subPackages = [ "cmd/modern" ];
+
+  preCheck = ''
+    export HOME=$(mktemp -d)
+  '';
+
+  nativeBuildInputs = [ installShellFiles ];
+
+  postInstall = ''
+    mv $out/bin/modern $out/bin/sqlcmd
+
+    installShellCompletion --cmd sqlcmd \
+      --bash <($out/bin/sqlcmd completion bash) \
+      --fish <($out/bin/sqlcmd completion fish) \
+      --zsh <($out/bin/sqlcmd completion zsh)
+  '';
+
+  passthru.tests.version = testers.testVersion {
+    package = sqlcmd;
+    command = "sqlcmd --version";
+    inherit version;
+  };
+
+  meta = with lib; {
+    description = "A command line tool for working with Microsoft SQL Server, Azure SQL Database, and Azure Synapse";
+    homepage = "https://github.com/microsoft/go-sqlcmd";
+    changelog = "https://github.com/microsoft/go-sqlcmd/releases";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ratsclub ];
+  };
+}
```

```suggestion
  meta = {
    description = "A command line tool for working with Microsoft SQL Server, Azure SQL Database, and Azure Synapse";
    homepage = "https://github.com/microsoft/go-sqlcmd";
    changelog = "https://github.com/microsoft/go-sqlcmd/releases";
    license = lib.licenses.mit;
    maintainers = [ lib.maintainers.ratsclub ];
  };
}
```

There is a problem when using `with` for statical analysis, so as a rule of thumb, avoid `with`, specially when nested (like `with lib; with maintainers;`
  - More information: https://github.com/atorres1985-contrib/nixpkgs/commit/451dfa6d84586b5cb83e6a7254689dff66b4523a